r-rse/course-design.md

```diff
@@ -0,0 +1,135 @@
+    - Is adequately structured to make package management and maintanence easier
+    (e.g. with Issue Templates, Project Boards, Labels, contributing guidelines)
+    
+- **Teaching/learning activities**:
+    - In class portion: 
+        - Some lectures for those material that aren't coding heavily
+        (e.g. for lessons to involve community, being a good community member)
+        - Live coding (with instructor)
+        - Exercises that expand on material covered with instructor that 
+        solve a provided problem, with or without scaffolding
+    - Out of class: 
+        - Group work/project creating an R package based on a set of (given?) problems
+        - Formative assessments like coding with immediate feedback on correct/incorrect
+        (via learnr package? Esp with new update to RStudio in version 1.3)
+        - Reading assignments?
+
+- **Assessment and feedback activities**:
+    - Deadlines for milestones to acheive for project, with instructor feedback on progress
+    - Out-of-class formative assessments based on maybe learnr tutorials?
+    - After completing in-class exercises, go over them as class?
+
+- Key resources to use for this:
+  - R packages: usethis, devtools, drake?, pkgdown, testthat, styler, rmarkdown
+  - Online books: [R Packages](https://r-pkgs.org/) 
+  (we differentiate from this amazing book by structuring everything around 
+  this being a course, with exercises, assessments, (group?) projects, 
+  instructions/comments/notes for instructors, etc)
+
+## Chapter outline
+
+Course chapters and content need to be strongly aligned with learning outcome/goal.
+Here is a draft outline to guide development.
+
+- Syllabus
+    - Overview of course
+    - Why to learn it
+    - What makes this course different
+
+- Folder and file structure (RStudio R Projects)
+    - Filepaths, very basics of the shell (in RStudio)
+    - Making use of the fs package for filesystem management
+    
+- Setting up an R package
+    - Using devtools, usethis for development
+        - e.g. `use_devtools()` and `use_usethis()`.
+    - Developing functions
+    - Making and using datasets
+
+- Function development in a package environment
+    - Dependency management
+    - Function documentation (with roxygen2), part 1
+        - To use `@examples` to help with creating function
+
+- (Idea) Additional features to make use of
+    - RStudio addins, snippets, rmarkdown templates, templates
+    - rstudioapi?
+
+- Version control: Using Git and GitHub as a sole user (part 1)
+    - Using Git in RStudio (standard add-commmit-history)
+    - Using Git in the Terminal of RStudio (moving in history with checkout,
+    creating branches, adding and updating remotes)
+    - Setting up GitHub for R package (make use of usethis), pushing and pulling
+
+- Checking correctness of code
+    - Unit test (with testthat)
+    - Code coverage (with covr)
+    - With devtools
```

Comment:
Are our learners already OK with debugging?  E.g. I'm thinking about Jenny Bryan's [Object of closure is not sub-settable keynote](https://rstudio.com/resources/rstudioconf-2020/object-of-type-closure-is-not-subsettable/), where unit tests are the last strategy she covers, do we assume the learners already know the earlier strategies?